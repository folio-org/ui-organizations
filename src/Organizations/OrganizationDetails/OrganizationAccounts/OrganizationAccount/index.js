export { default as OrganizationAccount } from './OrganizationAccount';
