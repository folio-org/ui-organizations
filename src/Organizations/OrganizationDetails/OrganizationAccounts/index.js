export * from './OrganizationAccount';
export { default as OrganizationAccounts } from './OrganizationAccounts';
