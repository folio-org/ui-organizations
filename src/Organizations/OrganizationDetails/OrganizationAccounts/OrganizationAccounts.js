import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { OrganizationAccount } from './OrganizationAccount';

const OrganizationAccounts = ({ accounts }) => {
  if (!accounts.length) {
    return (
      <p>
        <FormattedMessage id="ui-organizations.accounts.noAccountsAvail" />
      </p>
    );
  }

  return (
    <div>
      {
        accounts.map((account, idx) => (
          <OrganizationAccount
            key={idx}
            name={account.name}
            accountNo={account.accountNo}
            description={account.description}
            appSystemNo={account.appSystemNo}
            paymentMethod={account.paymentMethod}
            contactInfo={account.contactInfo}
            libraryCode={account.libraryCode}
            libraryEdiCode={account.libraryEdiCode}
            notes={account.notes}
          />
        ))
      }
    </div>
  );
};

OrganizationAccounts.propTypes = {
  accounts: PropTypes.arrayOf(PropTypes.object),
};

OrganizationAccounts.defaultProps = {
  accounts: [],
};

export default OrganizationAccounts;
