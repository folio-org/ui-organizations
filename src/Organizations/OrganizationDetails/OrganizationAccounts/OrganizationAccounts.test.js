import React from 'react';
import { render } from '@testing-library/react';

import OrganizationAccounts from './OrganizationAccounts';

const accounts = [{
  name: 'Amazon V',
  accountNo: 'A34sA',
  description: 'Use to buy',
  appSystemNo: '12567',
  paymentMethod: 'Cash',
  contactInfo: 'Max M',
  libraryCode: 'CDA',
  libraryEdiCode: 'ECDA',
  notes: 'Default account',
}];
const renderOrganizationAccounts = () => render(
  <OrganizationAccounts
    accounts={accounts}
  />,
);

describe('OrganizationAccounts', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', () => {
    const { asFragment } = renderOrganizationAccounts();

    expect(asFragment()).toMatchSnapshot();
  });
});
