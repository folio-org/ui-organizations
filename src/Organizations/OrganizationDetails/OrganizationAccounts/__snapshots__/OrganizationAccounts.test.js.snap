// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationAccounts should render correct structure 1`] = `
<DocumentFragment>
  <div
    data-test-accounts="true"
  >
    <div
      class="organizationAccount"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.name
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              Amazon V
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.accountNumber
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              A34sA
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.description
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              Use to buy
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.payable
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              12567
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
          data-test-payment-method="true"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.paymentMethod
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              stripes-acq-components.paymentMethod.cash
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.account.contactInfo
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              Max M
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.libraryCode
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              CDA
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.libraryEDICode
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              ECDA
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
        >
          <div
            class="kvRoot"
          >
            <div
              class="kvLabel"
            >
              ui-organizations.accounts.notes
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              Default account
            </div>
          </div>
        </div>
        <div
          class="col-xs-3"
          data-test-account-acq-units="true"
        >
          <div
            class="kvRoot"
            data-testid="acqUnits"
          >
            <div
              class="kvLabel"
            >
              stripes-acq-components.label.acqUnits
            </div>
            <div
              class="kvValue"
              data-test-kv-value="true"
            >
              <span>
                NoValue
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
